flink-java/src/test/java/org/apache/flink/api/java/utils/ParameterToolTest.java

```diff
@@ -174,9 +210,38 @@ public void testUnrequestedByte() {
 
 		// test repeated access
 		Assert.assertEquals(1, parameter.getByte("byte"));
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+	}
+
+	@Test
+	public void testUnrequestedByteWithDefaultValue() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"-byte", "1"});
+		Assert.assertEquals(Sets.newHashSet("byte"), parameter.getUnrequestedParameters());
+
+		// test parameter access
+		Assert.assertEquals(1, parameter.getByte("byte", (byte) 0));
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
+
+		// test repeated access
+		Assert.assertEquals(1, parameter.getByte("byte", (byte) 0));
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
 	}
 
 	@Test
+	public void testUnrequestedByteWithMissingValue() {
+		ParameterTool parameter = ParameterTool.fromArgs(new String[]{"-byte"});
+		Assert.assertEquals(Sets.newHashSet("byte"), parameter.getUnrequestedParameters());
+
+		exception.expect(RuntimeException.class);
+		exception.expectMessage("For input string: \"__NO_VALUE_KEY\"");
+
+		parameter.getByte("byte");
+		Assert.assertEquals(Collections.emptySet(), parameter.getUnrequestedParameters());
```

Comment:
I just tried myself with this little example:
```
	@Rule
	public final ExpectedException exception = ExpectedException.none();

	@Test
	public void testExpectedException() {
		exception.expect(RuntimeException.class);
		if (true) {
			throw new RuntimeException();
		}

		throw new NullPointerException();
	}
```

as expected, the `NullPointerException` is **not** thrown since the test already ends with the `RuntimeException`